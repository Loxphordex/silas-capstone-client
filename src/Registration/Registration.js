import React from 'react'
import { Link } from 'react-router-dom'
import ApiServices from '../services/api-service'
import history from '../history'
import './Registration.css'

export default class Registration extends React.Component {

  handleSubmit = (event) => {
    event.preventDefault()
    const { username, password } = event.target

    ApiServices.postUser(username.value, password.value)
      .then(user => {
        username.value=''
        password.value=''
        history.push('/entry')
      })
      .then(user => {
        this.props.clearError()
        this.props.history.history.push('/entry')
      })
      .catch(err => this.props.handleError(err))
  }

  render() {
    return (
      <div className='registration'>
        
        <nav role='navigation'>
          <h1 onClick={() => this.props.history.history.push('/')}>Quoter</h1>
        </nav>

        <h2>Register for an Account</h2>

        <div id='register-error' className='error'>
          {this.props.stateError && this.props.stateError}
        </div>

        <form id='register-form' onSubmit={(event) => this.handleSubmit(event)}>

          <label htmlFor='username'>Username</label>
          <input type='text' id='username' name='username'></input>

          <label htmlFor='password'>Password</label>
          <input type='password' id='password' name='password'></input>

          <button className='pure-button' type='submit'>Submit</button>
        </form>

        <div id='reg-to-login'><Link to='/login'>Already have an account?</Link></div>
      </div>
    )
  }
}