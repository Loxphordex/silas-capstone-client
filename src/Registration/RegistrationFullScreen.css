@media only screen and (min-width: 600px) {
  .registration input {
    font-size: 16px;
  }

  .register-main {
    margin: auto 20%;
  }

  .register-form {
    width: 320px;
  }
}