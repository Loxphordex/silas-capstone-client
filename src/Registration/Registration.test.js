import React from 'react';
import Registration from './Registration'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Registration Page', () => {
    it('renders the complete page', () => {
    const wrapper = shallow(<Registration />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})