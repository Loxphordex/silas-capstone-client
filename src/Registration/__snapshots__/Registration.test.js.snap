// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Page renders the complete page 1`] = `
<div
  className="registration"
>
  <nav
    role="navigation"
  >
    <h1
      className="pointer"
      onClick={[Function]}
    >
      Quoter
    </h1>
  </nav>
  <div
    id="register-main"
  >
    <h2>
      Register for an Account
    </h2>
    <div
      className="error"
      id="register-error"
    />
    <form
      id="register-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        type="password"
      />
      <button
        className="pure-button"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      id="reg-to-login"
    >
      <Link
        to="/login"
      >
        Already have an account?
      </Link>
    </div>
  </div>
</div>
`;
